lib/serializer/BinaryDeserializer.cpp

```diff
@@ -24,9 +24,8 @@ CLoadFile::CLoadFile(const boost::filesystem::path & fname, int minimalVersion)
 	openNextFile(fname, minimalVersion);
 }
 
-CLoadFile::~CLoadFile()
-{
-}
+///This destructor should be located here to avoid side effects
```

Comment:
What side effects? Please describe reason in comment in code

Reply:
Side effects is always same - something stops compiling or linking, because destructor implicitly requires knowledge about something included only in *.cpp. I need to explain this in each destructor?

Reply:
`//must be instantiated in .cpp file for access to complete types of all member fields`

Reply:
Problem is "side effect" is too generic. Could be some weird optimization issue on some platform, could be even compiler bug.